dbus.bbappend: Remove patch messing with readdir

gunnarx opened the pull request:
Let's test if this builds...

Commit:
dbus.bbappend: Remove patch for readdir/readdir_r

A patch was applied previously to make dbus 1.10 compile correctly in
the presence of the readdir_r call (and only for qemux86, 32-bit).

The update to poky/sumo now builds dbus 1.12 Between 1.10 and 1.12 the
code has moved and more importantly the readdir_r seems to have been
changed back to readdir, so the patch cannot apply.  Let's try removing
the patch.

The single patch however mixed different things.  Testing shows that
the signed/unsigned mismatch is still a problem (warning which is
treated as an error) but only on 32-bit compilation. This part of the
patch is therefore kept for now.   I have not deeply analyzed the effect
of this cast, just keeping what was already there.

[BASE-63] master fails application of dbus patch for qemux86 target

Signed-off-by: Gunnar Andersson <gandersson@genivi.org>

frznlogic commented:
That patch have pissed me off quite a bit... I'd love if they can be removed :)

gunnarx commented:
The patch messes with a few other things than readdir, but those changes seem not that useful.  

I fixed the spelling error in my message, but I won't push a cleanup until the first build results comes back.

gunnarx commented:
Hmm... PRs  not build-tested against x86 32-bit.  Added a pipeline, just to get this tested now.

frznlogic commented:
You have the same build issues I ran into locally. There is something with the optimization flags vs some internal flags. I didn't figure out how to solve it as I didn't have time :(

Commit:
dbus: adjust "blacklist" for hardening flags

Various component don't compile well with the added so called "hardening
flags".

The FORTIFY_SOURCE feature requires optimization to be enabled (-O1 or
-O2 depending...) It seems it would be better to force optimization on,
than to disable the "hardening flags" for these, but disabling them
seems standard practice.  (You are no worse off than before "hardening
flags" were added.  And all other components benefit). See
c5db7bafd3c837e, but also note that poky distro includes disabling these
for many components already.

There is a remaining unused-variable warning=error, which can be
squelched for the moment.  Later on, it ought to be restored.

I would think setting the right compiler flags should eventually be done
in the standard poky recipe (or possibly even, component upstream).

Signed-off-by: Gunnar Andersson <gandersson@genivi.org>

gunnarx commented:
Yes but it seems to have nothing do with the patch, so it should exist also on master (for 32 bit build)?  Anyway, I'll push a blacklist modification just to confirm if the error goes away.

gunnarx commented:
Latest failure: dbus-message.c:4372:36: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]

And of course I had to eat my words.  Looks like one part of the patch was actually necessary...  Apparently the sign mismatch problem is not triggered in a 64 bit build.  Patch needs to be refreshed if we keep it.

Oh and I guess the "blacklisting" of the hardening flags worked...

gunnarx commented:
- Adjusted patch again to include the integer cast that was originally proposed.  I have not checked that deeply - just keeping what was there already.

- Then there was an unused variable also which was caught by -Werror to become a failure.  I temporarily disabled the treating of this as an error for dbus compilation.  I checked and in later D-Bus versions this has been fixed by moving the declaration inside the #ifdef where it is conditionally used.  So when dbus component is updated in layers later on, then -Werror could be enabled again for unused vars (but that is a fairly strict setting to have...)

Guess it's ready for review now.

gunnarx commented on meta-ivi/recipes-core-ivi/dbus/dbus/build_fix-ubunt1604-qemux86.patch:
```diff
@@ -30,15 +9,3 @@
      {
```
This part of the patch is unchanged..., the rest is removed.

gunnarx commented:
I had a quick chat with frznlogic on this, so I think I can go ahead and merge it.  It should fix the build-error for master and other PRs so even if the solution might not be perfect, it's worthwhile to merge it now.

gunnarx commented on meta-ivi/recipes-core-ivi/dbus/dbus/build_fix-ubunt1604-qemux86.patch:
```diff
@@ -30,15 +9,3 @@
-             {
-             case '\n':
-               if (!delimit_token (&current_token, &retval, error))
-+		{
```
Skipped this since it serves no purpose.  This is a  style change, that should either just be accepted as it is, or changed upstream, not here.
